### ISSUE TITLE
Show correct line numbers

### ISSUE THREAD

**Issue by SuperSandro2000** [hadolint/hadolint#285]
- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Steps to reproduce the behavior

```
RUN apt-get update \
  .......
  && curl -Lso Lavalink.jar $(curl -s https://api.github.com/repos/Cog-Creators/Red-DiscordBot/releases | jq -r '.[0].assets | .[0].browser_download_url') \
```
This triggers SC2046 on the line of RUN and not on the line of the issue that is the curl line.

### Expected behavior

Give me the exact line of the issue.

### Actual behavior

Start line of the docker command it returned.

**Comment by lorenzo** [comment]
Thanks for your interest. This is a very difficult feature to implement, because of the rules of breaking lines in dockerfiles. I don't anticipate cracking the problem anytime soon, although I have some wild ideas on how to do it.

**Comment by SuperSandro2000** [comment]
I am not familiar with haskel at all but how I would approach it in other programming languages (Hope this helps you):
- get the beginning of the matched string
- count all LF characters until you hit one of the docker commands (RUN, COPY, etc.)
- Add the LF-count to the now already known Line number
Should work on Linux and Windows with CR LF.
Or do I miss something?
